src/app/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts

```diff
@@ -250,7 +250,14 @@ export const useLocalArtworkFilter = (artworks: any[]) => {
         },
       ])
     )
+  }
+  useEffect(() => {
+    if (artworksList) {
+      initLocalArtworkFilter(artworksList)
+    }
   }, [])
```

Comment:
I don't think we need to return `reInitializeLocalArtworkFilter`. We could just move the effect from `MyCollection` to `useLocalArtworkFilter`. 

```suggestion
  useEffect(() => {
    if (artworksList) {
      initLocalArtworkFilter(artworksList)
    }
  }, [artworks])
```

Reply:
Moving the effect to useLocalArtworkFilter will make debugging / reproducing bugs related to Local Filters really really difficult.

Reply:
I'm not sure if I can follow. Why does it make debugging harder? I only see the benefit from having all logic related to the filters inside of the hook instead of spreading it between the hook and the component.

Reply:
@olerichter00 Well, I should have said "...really really difficult `for me`". 😂 
1. If we move that effect, the initialization will still be done once and we'd be back to square 1. Because we will need an effect that depends on artwork for the component to rerender for any reinitialization to occur.
2. I'm always for letting components use methods/hooks however they like and whenever they like. I'd rather a component initialize with artworks (or not) or initialize following an arbitrary event or even initialize with extra objects we may not be displaying. Helps me narrow down what is happening when and where.

Reply:
Oh @olerichter00 I see your point now. Deleting my previous comment....

Reply:
Well, I think having components able to initialise localArtworkFilter when and how they want would always help narrow down what's happening where, how and when. I'm always against abstracting so much away from components.  Happy to change it if you insist.

Reply:
I can't think of another case where the filters need to be updated. They completely depend on the list artworks.